### PR TITLE
WRO-11485: Update package dependencies to fix security vulnerabilities

### PR DESCRIPTION
**Pull request by andreitirlalge**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Updated dependencies for Moonstone, Sandstone, Agate and UI.
Migrated react-router-dom from version from 5.2.0 to 6.4.2.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Dependencies now use the latest version.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-11485

### Comments
Enact-DCO-1.0-Signed-off-by: Andrei Tirla andrei.tirla@lgepartner.com

### CODE REVIEW

**Review by hong6316 — COMMENTED** [review]
Could you consider `"readdirp": "^3.6.0"` in the root directory? like WRO-460?(Please make a new ticket).

Except this, these changes are LGTM.

**Review by hong6316 — APPROVED** [review]
OK :)

File: moonstone/all-samples/src/components/SampleItem/SampleItem.js [hunk 1]
```diff
@@ -1,14 +1,11 @@
-/* eslint-disable react/jsx-no-bind */
-
 import Item from '@enact/moonstone/Item';
 import PropTypes from 'prop-types';
+import {useCallback} from "react";
```
Comment (MikyungKim) [inline moonstone/all-samples/src/components/SampleItem/SampleItem.js:3]:
```suggestion
import {useCallback} from 'react';
```

File: sandstone/all-samples/src/App/App.js [hunk 2]
```diff
@@ -2,7 +2,8 @@ import kind from '@enact/core/kind';
 import Scroller from '@enact/sandstone/Scroller';
 import ThemeDecorator from '@enact/sandstone/ThemeDecorator';
 import PropTypes from 'prop-types';
-import {HashRouter, Route, StaticRouter} from 'react-router-dom';
+import {HashRouter, Route, Routes, useNavigate} from 'react-router-dom';
+import {StaticRouter} from "react-router-dom/server";
```
Comment (MikyungKim) [inline sandstone/all-samples/src/App/App.js:6]:
```suggestion
import {StaticRouter} from 'react-router-dom/server';
```

File: sandstone/tutorial-hello-enact/package.json [hunk 3]
```diff
@@ -27,12 +27,12 @@
...
-    "@enact/i18n": "^4.5.0",
-    "@enact/sandstone": "^2.5.0",
-    "@enact/spotlight": "^4.5.0",
-    "@enact/ui": "4.5.0",
-    "ilib": "^14.14.0",
+    "@enact/core": "^4.5.2",
+    "@enact/i18n": "^4.5.2",
+    "@enact/sandstone": "^2.5.5",
+    "@enact/spotlight": "^4.5.2",
+    "@enact/ui": "4.5.2",
```
Comment (MikyungKim) [inline sandstone/tutorial-hello-enact/package.json:34]: Why is this version pinned?

File: ui/all-samples/src/App/App.js [hunk 4]
```diff
@@ -1,13 +1,15 @@
 import kind from '@enact/core/kind';
 import PropTypes from 'prop-types';
 import {routes} from  '../index';
+import {useNavigate} from 'react-router-dom';
 import SampleItem from '../components/SampleItem';
 import Scroller from '@enact/ui/Scroller';
```
Comment (MikyungKim) [inline ui/all-samples/src/App/App.js:6]: Please reorder imports.

File: agate/all-samples/src/App/App.js [hunk 5]
```diff
@@ -2,7 +2,8 @@ import Scroller from '@enact/agate/Scroller';
 import ThemeDecorator from '@enact/agate/ThemeDecorator';
 import kind from '@enact/core/kind';
 import PropTypes from 'prop-types';
-import {HashRouter, Route, StaticRouter} from 'react-router-dom';
+import {HashRouter, Route, Routes, useNavigate} from 'react-router-dom';
+import {StaticRouter} from "react-router-dom/server";
```
Comment (MikyungKim) [inline agate/all-samples/src/App/App.js:6]:
```suggestion
import {StaticRouter} from 'react-router-dom/server';
```

File: moonstone/all-samples/src/App/App.js [hunk 6]
```diff
@@ -2,7 +2,8 @@ import kind from '@enact/core/kind';
 import MoonstoneDecorator from '@enact/moonstone/MoonstoneDecorator';
 import Scroller from '@enact/moonstone/Scroller';
 import PropTypes from 'prop-types';
-import {HashRouter, Route, StaticRouter} from 'react-router-dom';
+import {HashRouter, Route, Routes, useNavigate} from 'react-router-dom';
+import {StaticRouter} from "react-router-dom/server";
```
Comment (MikyungKim) [inline moonstone/all-samples/src/App/App.js:6]:
```suggestion
import {StaticRouter} from 'react-router-dom/server';
```

**Review by MikyungKim — CHANGES_REQUESTED** [review]
Please find my comments.

File: sandstone/tutorial-hello-enact/package.json (hunk 3 above)
Comment (andreitirlalge) [reply sandstone/tutorial-hello-enact/package.json:34]: to be honest, i did not notice it, but it was here for a while. Thank you for noticing. I will correct it

File: agate/all-samples/src/App/App.js [hunk 7]
```diff
@@ -18,19 +19,21 @@ import css from './App.module.less';
 
 const NavigationMenu = kind({
 	name: 'NavigationMenu',
+	functional: true,
```
Comment (daniel-stoian-lgp) [inline agate/all-samples/src/App/App.js:22]:
```suggestion

	functional: true,
```

File: moonstone/all-samples/src/App/App.js [hunk 8]
```diff
@@ -25,19 +26,21 @@ import css from './App.module.less';
 
 const NavigationMenu = kind({
 	name: 'NavigationMenu',
+	functional: true,
```
Comment (daniel-stoian-lgp) [inline moonstone/all-samples/src/App/App.js:29]:
```suggestion

	functional: true,
```

File: sandstone/all-samples/src/App/App.js [hunk 9]
```diff
@@ -24,19 +25,21 @@ import css from './App.module.less';
 
 const NavigationMenu = kind({
 	name: 'NavigationMenu',
+	functional: true,
```
Comment (daniel-stoian-lgp) [inline sandstone/all-samples/src/App/App.js:28]:
```suggestion

	functional: true,
```

File: ui/all-samples/src/App/App.js [hunk 10]
```diff
@@ -3,11 +3,13 @@ import PropTypes from 'prop-types';
 import {routes} from  '../index';
 import SampleItem from '../components/SampleItem';
 import Scroller from '@enact/ui/Scroller';
+import {useNavigate} from 'react-router-dom';
 
 import css from './App.module.less';
 
 const App = kind({
 	name: 'App',
+	functional: true,
```
Comment (daniel-stoian-lgp) [inline ui/all-samples/src/App/App.js:12]:
```suggestion

	functional: true,
```

**Review by daniel-stoian-lgp — COMMENTED** [review]
added missing empty lines

**Review by daniel-stoian-lgp — APPROVED** [review]

**Review by MikyungKim — COMMENTED** [review]
Just out of curiousity, is the image showing well in your PC for all-samples > PatternSinglePanel?
I'm not sure but my result showing a gray image.
image
🤔

**Comment by daniel-stoian-lgp** [comment]
> Just out of curiousity, is the image showing well in your PC for all-samples > PatternSinglePanel? I'm not sure but my result showing a gray image. image 🤔

Images in All-samples sample fot all theme libraries never loaded on our PCs. We did an analysis and it is trying to load them from a  path inside all-samples folder, but they are located in the other samples' folder. 
One solution could be to duplicate those images or check how to do configure routing for this specific case, but in another user story :) What do you think?

**Comment by MikyungKim** [comment]
🤔 I was confused since the built one uploaded to our server shows images well. Let's take a look at this in another ticket.

File: agate/all-samples/package.json [hunk 11]
```diff
@@ -29,20 +29,20 @@
...
   "dependencies": {
-    "@enact/agate": "^2.0.0",
-    "@enact/core": "^4.5.0",
-    "@enact/i18n": "^4.5.0",
-    "@enact/spotlight": "^4.5.0",
-    "@enact/ui": "^4.5.0",
-    "@enact/webos": "^4.5.0",
-    "@reduxjs/toolkit": "^1.8.1",
-    "ilib": "^14.14.0",
+    "@enact/agate": "^2.0.1",
```
Comment (MikyungKim) [inline agate/all-samples/package.json:32]: Oh, we recently updated agate to 2.0.2.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**